### PR TITLE
Remove the need for getters and setters from demo and use immutable operations

### PR DESCRIPTION
**Pull request by AndyTitu**
This PR leaves out defining the `setField` and `getField` helper functions and uses immutable iterator functions to achieve the same result.

### CODE REVIEW

File: rotation-service/index.mjs [hunk 1]
```diff
@@ -26,47 +26,31 @@ export const handler = async (event) => {
...
   const newAPIKey = await newAPIKeyRequest;
 
-  setField(item, "rolledKeySID", getField(item, "sid"));
-  setField(item, "sid", newAPIKey.sid);
-  setField(item, "secret", newAPIKey.secret);
+  let newItem = {
+    ...item,
+    fields: item.fields.map((f) => {
+      if (f.title == "rolledKeySID") {
+        return { ...f, value: item.fields.find(f => f.title == "sid").value};
```
Comment (Marton6) [inline rotation-service/index.mjs]: To simplify this, could we move out the sid value into a const variable?

File: rotation-service/index.mjs [hunk 2]
```diff
@@ -26,47 +26,31 @@ export const handler = async (event) => {
...
-  const rolledKeySID = getField(item, "rolledKeySID");
+  const rolledKeySID = item.fields.find(f => f.title == "rolledKeySID").value;
   twilioClient.keys(rolledKeySID).remove();
 
   const newAPIKey = await newAPIKeyRequest;
 
-  setField(item, "rolledKeySID", getField(item, "sid"));
-  setField(item, "sid", newAPIKey.sid);
-  setField(item, "secret", newAPIKey.secret);
+  let newItem = {
```
Comment (Marton6) [inline rotation-service/index.mjs]: **Nit:** could we call this `updatedItem` instead? To make it super clear we're not creating a new item, but updating an existing one

Comment (AndyTitu) [reply rotation-service/index.mjs]:
```suggestion
  let updatedItem = {
```

File: rotation-service/index.mjs [hunk 3]
```diff
@@ -26,47 +26,31 @@ export const handler = async (event) => {
...
+      } else if (f.title == "secret") {
+        return { ...f, value: newAPIKey.secret};
+      } else {
+        return f;
+      }
+    }),
+  };
 
-  await client.items.put(item);
+  await client.items.put(newItem);
```
Comment (AndyTitu) [inline rotation-service/index.mjs]:
```suggestion
  await client.items.put(updatedItem);
```

File: rotation-service/index.mjs (hunk 1 above)
Comment (AndyTitu) [reply rotation-service/index.mjs]: done

File: rotation-service/index.mjs [hunk 4]
```diff
@@ -26,47 +26,32 @@ export const handler = async (event) => {
...
+        return { ...f, value: newRolledKeySID};
+      } else if (f.title == "sid") {
+        return { ...f, value: newAPIKey.sid};
+      } else if (f.title == "secret") {
+        return { ...f, value: newAPIKey.secret};
+      } else {
+        return f;
+      }
+    }),
+  };
```
Comment (SimonBarendse) [inline rotation-service/index.mjs]:
```suggestion
  const previousKeySID = item.fields.find(f => f.title == "sid").value

  let updatedItem = {
    ...item,
    fields: item.fields.map((f) => {
      if (f.title == "rolledKeySID") {
        return { ...f, value: previousKeySID};
      } else if (f.title == "sid") {
        return { ...f, value: newAPIKey.sid};
      } else if (f.title == "secret") {
        return { ...f, value: newAPIKey.secret};
      } else {
        return f;
      }
    }),
  };
```

To simplify the language a bit: the newly rolled key is "just" the previous key.

**Review by SimonBarendse — APPROVED** [review]

**Pull request merged**